network/tls_authentication_test.go

```diff
@@ -1,10 +1,76 @@
 package network
 
 import (
+	"bytes"
+	"crypto/rand"
+	"crypto/sha512"
 	"crypto/x509"
+	"encoding/hex"
+	"encoding/pem"
+	"io/ioutil"
+	"os"
+	"path/filepath"
 	"testing"
 )
 
+// getWorkingDirectory expects that tests started from root of source code and accessible tests/ssl folder, otherwise call t.Fatal otherwise
+func getWorkingDirectory(t *testing.T) string {
+	workingDirectory, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+	sslTestFolder := filepath.Join(workingDirectory, "tests/ssl")
+	info, err := os.Lstat(sslTestFolder)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !info.IsDir() {
+		t.Fatalf("'%s' is not directory as expected\n", sslTestFolder)
+	}
+	return workingDirectory
+}
+
+func getAcraWriterTestx509Certificate(t *testing.T) *x509.Certificate {
+	certPath := filepath.Join(getWorkingDirectory(t), "tests/ssl/acra-writer/acra-writer.crt")
+	certificateData, err := ioutil.ReadFile(certPath)
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
Hm.... Judging by the usage pattern of `getWorkingDirectory()`, why not convert it into some
```go
func readTestCertificateData(t *testing.T, path string) []byte {
	workingDirectory, err := os.Getwd()
	if err != nil {
		t.Fatal(err)
	}
	sslTestFolder := filepath.Join(workingDirectory, "tests/ssl")
	info, err := os.Lstat(sslTestFolder)
	// The "err" is likely to be os.ErrNotExist if the directory is mismatched,
	// so I guess both failures can be merged...
	if err != nil || !info.IsDir() {
		t.Fatalf("Current working directory '%s' is not Acra root directory", workingDirectory)
	}
	data, err := ioutil.ReadFile(filepath.Join(sslTestFolder, path))
	if err != nil {
		t.Fatal(err)
	}
	return data
}
```
which does all of the above checks and is easier to comprehend from tests?

I doubt that we'd need to check current working directory for anything else.

Reply:
first version was like that until I started tests from IDE and working directory automatically configured to `network/ ` when I run tests from network package) And I thought that other developers may meet these problems too and will have to change working directory every time. So decided to simplify it for all